lib/daru/vector.rb

```diff
@@ -199,19 +199,62 @@ def initialize source, opts={}
     #   # For vectors employing hierarchial multi index
     #
     def [](*input_indexes)
-      # Get a proper index object
-      indexes = @index[*input_indexes]
+      # Get array of positions indexes
+      positions = @index.pos(*input_indexes)
```

Comment:
Great idea. I didn't think about it. Now that I tried it, it almost worked. There's problem with multi index though. The following scenario illustrates it:

```ruby
[1] pry(main)> idx = Daru::MultiIndex.from_tuples [[:a, :one], [:a, :two]]
=> #<Daru::MultiIndex(2x2)>
   a one
     two
[2] pry(main)> dv = Daru::Vector.new [1, 2], index: idx
=> #<Daru::Vector(2)>
   a one   1
     two   2
[3] pry(main)> dv[:a]
=> #<Daru::Vector(2)>
 one   1
 two   2
[4] pry(main)> dv[0, 1]
=> #<Daru::Vector(2)>
   a one   1
     two   2
```